trlx/sweep.py

```diff
@@ -1,83 +1,310 @@
+                max_runs_to_show=100,
+                plot_type="line",
+                font_size="auto",
+                legend_position="north",
+            )
+        )
+
+    report.blocks = report.blocks + [
+        wb.H1(text="Metrics"),
+        wb.PanelGrid(
+            panels=line_plot_panels,
+            runsets=[wb.Runset(project=project_name).set_filters_with_python_expr(f'group == "{group_name}"')],
+        ),
+    ]
+
+    if best_config:
+        best_config = best_config["train_loop_config"]
+        config = {}
+        for name, value in best_config.items():
+            *layers, var = name.split(".")
+            if layers:
+                d = config.setdefault(layers[0], {})
+                for layer in layers[1:]:
+                    d = d.setdefault(layer, {})
+                d[var] = value
+
+        report.blocks = report.blocks + [
+            wb.H1(text="Best Config"),
+            wb.CodeBlock(code=[json.dumps(config, indent=4)], language="json"),
+        ]
+
+    report.save()
+    print(report.url)
 
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
-    parser.add_argument("script", type=str, help="Path to the script")
+    parser.add_argument("script", type=str, help="Path to the example script")
     parser.add_argument(
         "--config",
         type=str,
         required=True,
         help="The config file defining the param_space.",
     )
-    parser.add_argument("--num-cpus", type=int, default=4, help="Number of CPUs to use per exp.")
-    parser.add_argument("--num-gpus", type=int, default=1, help="Number of GPUs to use per exp.")
-    parser.add_argument("-y", "--assume-yes", action="store_true", help="Don't ask for confirmation")
+
+    parser.add_argument(
+        "--accelerate_config",
+        type=str,
+        required=False,
+        help="The default config file for the script.",
+    )
+    parser.add_argument("--num_gpus", type=int, default=1, help="Number of GPUs (workers) to use per trial.")
+    parser.add_argument("--num_cpus", type=int, default=4, help="Number of CPUs to use per GPU (worker).")
+    parser.add_argument("-y", "--assume_yes", action="store_true", help="Don't ask for confirmation")
     parser.add_argument(
-        "--server-address",
+        "--server_address",
         type=str,
         default=None,
         required=False,
         help="The address of server to connect to if using Ray Client.",
     )
 
-    args, _ = parser.parse_known_args()
+    args = parser.parse_args()
 
-    # Read config and parse it
     with open(args.config) as f:
         config = yaml.safe_load(f)
     tune_config = get_tune_config(config.pop("tune_config"))
     param_space = get_param_space(config)
+    column_names = list(param_space.keys())
+    target_metric = tune_config["metric"]
 
-    # Initialize Ray.
     if args.server_address:
         ray.init(address=f"ray://{args.server_address}")
     else:
         ray.init()
```

Comment:
Running the example script from the `README.md` within a singe node 8xA100 instance (CoreWeave) I get `ConnectionError`s of the kind:
```sh
ConnectionError: Could not read 'session_name' from GCS. Did GCS start successfully?
```
(obviously not using GCS)

Passing the `local` flag to this init (`ray.init("local")`) seems to address the issue. So, I'm wondering if maybe we should include a CLI arg for such use cases or simply update the README to tell users to start a ray instance first and then run the command. Has any PR author run into the `ConnectionError` on the most recent commit?

Note: This is in a fresh env from this branch.

Reply:
`ray.init()` just by itself should work fine, and it will automatically start a Ray cluster composed of just the current instance if one is not running already. Are you setting the `server_address` arg?

Reply:
I no longer have this issue. There have been some network issues with our cluster recently so I'm going to chalk it up to that. Note that I did not set the `server_address` arg for what it's worth. 
Re GCS:  I quickly assumed it was Google Cloud related, thank you for clarifying 🙏

Reply:
> I no longer have this issue
@jon-tow How was it resolved for you? I experience the same thing right now, never encountering it before. Something must messing with ray autodiscovery, either starting a ray cluster by hand or forcing `local` does work however.

Reply:
I'm not quite sure. I tried a new node and it was then able to auto-launch locally 😅